deeppavlov/models/kbqa/query_generator.py

```diff
@@ -0,0 +1,597 @@
+
+                ex_rels_2 = []
+                for entity in entity_ids[0][:self.entities_to_leave]:
+                    for rel in top_rels[:self.rels_to_leave]:
+                        objects_mid = self.wiki_parser("objects", "forw", entity, rel, type_of_rel="direct")
+                        objects_mid += self.wiki_parser("objects", "backw", entity, rel, type_of_rel="direct")
+                        if len(objects_mid) < 10:
+                            for obj in objects_mid:
+                                ex_rels_2 += self.wiki_parser("rels", "forw", obj, type_of_rel="direct")
+
+                ex_rels_2 = list(set(ex_rels_2))
+                scores_2 = self.rel_ranker([question], ex_rels_2, False) if self.return_answers \
+                                    else self.rel_ranker(question, ex_rels_2)
+                top_rels_2 = [score[0] for score in scores_2]
+                if self.debug:
+                    log.debug(f"top scored second rels: {top_rels_2}")
+
+                for entity in entity_ids[0][:self.entities_to_leave]:
+                    for rel in top_rels[:self.rels_to_leave]:
+                        objects = self.wiki_parser("objects", "forw", entity, rel, type_of_rel="direct")
+                        objects += self.wiki_parser("objects", "backw", entity, rel, type_of_rel="direct")
+                        if objects:
+                            candidate_outputs.append([rel, objects[0]])
+
+                for entity in entity_ids[0][:self.entities_to_leave]:
+                    for rel_1 in top_rels[:self.rels_to_leave]:
+                        objects_mid = self.wiki_parser("objects", "forw", entity, rel_1, type_of_rel="direct")
+                        objects_mid += self.wiki_parser("objects", "backw", entity, rel_1, type_of_rel="direct")
+                        if objects_mid and len(objects_mid) < 10:
+                            for obj in objects_mid:
+                                for rel_2 in top_rels_2[:self.rels_to_leave]:
+                                    objects = self.wiki_parser("objects", "forw", obj, rel_2, type_of_rel="direct")
+                                    if objects:
+                                        candidate_outputs.append([rel_1, rel_2, objects[0]])
+
+        if len(entity_ids) == 2:
+            ent_combs = self.make_entity_combs(entity_ids)
+            if rels_from_template is not None:
+                candidate_outputs = self.from_template_two_ent(ent_combs, rels_from_template)
+
+            else:
+                for ent_comb in ent_combs:
+                    ex_rels = []
+                    ex_rels += self.wiki_parser("rels", "forw", ent_comb[1], type_of_rel="direct")
+                    ex_rels += self.wiki_parser("rels", "backw", ent_comb[1], type_of_rel="direct")
+
+                    ex_rels = list(set(ex_rels))
+                    scores = self.rel_ranker([question], ex_rels, False) if self.return_answers \
```

Comment:
`RelRankerInfer.__call__` method receives only two arguments, `False` is unexpected.